File: Parallel_And_Distributed_Computing/Dijkstra’sAlgorithm/Test/ThreadBig.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Dijkstras_Algorithm;

namespace Test
{
    [TestClass]
    public class ThreadBig
    {
        [TestMethod]
        public void SizeSmall_ThreadBig()
        {
            new Dijkstra(new Graph(Tester.SMALL_SIZE).Generate()).Run(from: 0, threadAmount: Tester.BIG_THREAD);
        }
        [TestMethod]
        public void SizeMiddle_ThreadBig()
        {
            new Dijkstra(new Graph(Tester.MIDDLE_SIZE).Generate()).Run(from: 0, threadAmount: Tester.BIG_THREAD);
        }
        [TestMethod]
        public void SizeBig_ThreadBig()
        {
            new Dijkstra(new Graph(Tester.BIG_SIZE).Generate()).Run(from: 0, threadAmount: Tester.BIG_THREAD);
        }
    }
}

File: Parallel_And_Distributed_Computing/Dijkstra/Test/ThreadBig.cpp
#include "stdafx.h"
#include "CppUnitTest.h"

using namespace Microsoft::VisualStudio::CppUnitTestFramework;
using namespace CUDA;

namespace Test
{
	TEST_CLASS(ThreadBig)
	{
	public:

		TEST_METHOD(small_size_thread_big)
		{
			Graph<double> *graph = new Graph<double>(SMALL_SIZE);
			Dijkstra<double> *dijkstra = new Dijkstra<double>(graph);
			dijkstra->run(0, BIG_THREAD);
			delete graph;
			delete dijkstra;
		}
		TEST_METHOD(middle_size_thread_big)
		{
			Graph<double> *graph = new Graph<double>(MIDDLE_SIZE);
			Dijkstra<double> *dijkstra = new Dijkstra<double>(graph);
			dijkstra->run(0, BIG_THREAD);
			delete graph;
			delete dijkstra;
		}
		TEST_METHOD(big_size_thread_big)
		{
			Graph<double> *graph = new Graph<double>(BIG_SIZE);
			Dijkstra<double> *dijkstra = new Dijkstra<double>(graph);
			dijkstra->run(0, BIG_THREAD);
			delete graph;
			delete dijkstra;
		}

	};
}